src/test/java/com/excelsiorjet/api/tasks/JetProjectTest.java

```diff
@@ -0,0 +1,132 @@
+package com.excelsiorjet.api.tasks;
+
+import com.excelsiorjet.api.ExcelsiorJet;
+import com.excelsiorjet.api.tasks.config.ClasspathEntry;
+import com.excelsiorjet.api.tasks.config.DependencySettings;
+import com.excelsiorjet.api.tasks.config.ProjectDependency;
+import com.excelsiorjet.api.util.Txt;
+import com.excelsiorjet.api.util.Utils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.excelsiorjet.api.tasks.JetBuildTaskTest.mockUtilsClass;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(PowerMockRunner.class)
+public class JetProjectTest {
+
+    @Test
+    public void testExternalDirectoryNotPackedValidation() throws Exception {
+        File extDepDirSpy = Tests.dirSpy(Tests.projectDir.resolve("externalDir").toString());
+        Mockito.when(extDepDirSpy.isDirectory()).thenReturn(true);
+
+        JetProject project = Tests.testProject(ApplicationType.PLAIN).
+                dependencies(singletonList(DependencyBuilder.testExternalDependency(extDepDirSpy).pack(PackingType.ALL).asDependencySettings()));
+
+        try {
+            project.validate(Mockito.spy(new ExcelsiorJet(null, null)), false);
+            fail("JetTaskFailureException expected");
+        } catch (JetTaskFailureException e) {
+            assertEquals("External dependencies pointing to directory may have only \"none\" pack mode. Invalid dependency: \"/tmp/excelsior-jet-api-test/prj/externalDir\".", e.getMessage());
```

Comment:
There is no semantic meaning for full text, only its is matters while the text itself can be changed